### DISCUSSION TITLE
Provide Dockerfile for self build Image?

### DISCUSSION THREAD

**Discussion by flomickl** [onthegomap/planetiler#714] [Q&A] [answered]
Hi all,

I am using docker/podman and everything is working like a charm to use planetiler.
Big thanks to all of you working on this cool project.

But I need to use a different base image due to some CVSS requirements on my side.
Is it possible to provide the Containerfile/Dockerfile that was used to create the ghcr.io/onthegomap/planetiler:latest image

 I can't find it in the project and I want to use it as a reference?

Currently, I don't know which packages are required apart from Java 21. Is there a list of all requirements (like jib-maven-plugin:3.4.0) and the entry point __cacert_entrypoint.sh?

There also seems to be a mistake because the Image was supposedly built 53 years ago;-)

```
REPOSITORY                                         TAG               IMAGE ID      CREATED       SIZE
ghcr.io/onthegomap/planetiler                      latest            6370b9ebbe05  53 years ago  367 MB
````

This is probably due to jib-maven-plugin:3.4.0 when I look at it via docker/podman inspect:
```
    {
        "id": "6370b9ebbe05b8e6adf7e0630c9202b1b33e434601e1038503ec989d8dab9a79",
        "created": "1970-01-01T00:00:00Z",
        "CreatedBy": "jib-maven-plugin:3.4.0",
        "size": 6656,
        "comment": "jvm arg files",
        "Created": "1970-01-01T00:00:00Z"
    },
...
```

**Comment by msbarry** [comment]
Planetiler uses jib to build the docker image, you can change the base image here:

https://github.com/onthegomap/planetiler/blob/517851435d8e46587c7ada71750a75d2b9892505/planetiler-dist/pom.xml#L62-L64

If you want to make a custom docker image, you can also get the jar file into your image and just run the java commands from the "run without docker" instructions?

The date issue is one weird thing jib does that I haven't quite figured out how to fix - if you have any ideas let me know!

**Reply by flomickl** [reply]
Hm, I've never worked with jib before. How I start the process is my main question at the moment. I've had a look at https://github.com/GoogleContainerTools/jib, but I'm a bit overwhelmed with the amount of information. The structure in the pom file is (sometimes) clear to me, but I don't understand where the entrypoint is configured, for example.
After all the obstacles, I can certainly help out. I always write my container files separately.

**Comment by flomickl** [comment, marked as answer]
I created a Containerfile by myself using the ubi9 (RHEL) image and a non-root user.
Maybe this is helpful for other users as well.


Stored in my example as containerfile

```
FROM registry.access.redhat.com/ubi9/ubi-minimal

ARG PLANETILER_VERSION=0.7.0

USER root

RUN microdnf --setopt=install_weak_deps=0 --setopt=tsflags=nodocs install -y java-21-openjdk-devel wget shadow-utils && \
    rm -rf /var/cache/yum

# Set Java environment variables
ENV JAVA_HOME="/usr/lib/jvm/java-21"
ENV JAVA_VENDOR="openjdk"
ENV JAVA_VERSION="21"

# planetiler version
ENV PLANETILER_VERSION=${PLANETILER_VERSION}

#ROOTLESS USER AND GROUP
ENV CNB_USER_ID=1001
ENV CNB_GROUP_ID=1001

#default value
ENV JAVA_TOOL_OPTIONS="-Xmx1g"

RUN groupadd -g $CNB_GROUP_ID default && useradd -u $CNB_USER_ID default -g $CNB_GROUP_ID && \
    install -d -m 0700 -o $CNB_USER_ID -g $CNB_GROUP_ID /opt/planetiler && \
    install -d -m 0700 -o $CNB_USER_ID -g $CNB_GROUP_ID /data

# change to non-root user
USER $CNB_GROUP_ID

RUN echo "Downloading planetiler version ${PLANETILER_VERSION}" && \
    wget --output-document=/opt/planetiler/planetiler.jar -q https://github.com/onthegomap/planetiler/releases/download/v"$PLANETILER_VERSION"/planetiler.jar

WORKDIR /data

ENTRYPOINT ["java", "-jar", "/opt/planetiler/planetiler.jar"] 

```

This image can be built via docker /podman via

`docker build -t planetiler:0.7.0 --no-cache --progress=plain --build-arg PLANETILER_VERSION=0.7.0 -f containerfile .`

and can be run like in the example

`docker run -e JAVA_TOOL_OPTIONS="-Xmx3g" -v ./data":/data planetiler:latest --download --area=monaco`

**Reply by msbarry** [reply]
Great! Thanks for putting that together.

**Comment by flomickl** [comment]
@msbarry Do you know if it is native possible to set the arguments like `--download` `--area=monaco` and all the other arguments via environment variables without writing them behind the` java -jar planetiler.jar` statement?

**Reply by msbarry** [reply]
Yes! For any argument you set with `--arg-name=value` you can set it via an environmental variable `PLANETILER_ARG_NAME=value`, for example `PLANETILER_DOWNLOAD=true PLANETILER_AREA=monaco`
